oauthlib/oauth2/rfc6749/endpoints/resource.py

```diff
@@ -72,7 +74,19 @@ def verify_request(self, uri, http_method='GET', body=None, headers=None,
                                              self.default_token_type_handler)
         log.debug('Dispatching token_type %s request to %r.',
                   request.token_type, token_type_handler)
-        return token_type_handler.validate_request(request), request
+        is_valid = token_type_handler.validate_request(request)
+
+        # the validate_request() call is a chance for the client library to finalize the scopes on the request
```

Comment:
I'm ok with this living in `validate_request()` given the way the `request` is currently being populated within that code flow but if there's a clear desire to move it into a new `RequestValidator` method where this `claims` parameter parsing takes place let me know.